server/apps/tasks.py

```diff
@@ -260,4 +266,9 @@ def on_deleted(self, doc):
     def assign_user(self, item_id, user):
         return self.patch(item_id, {'task': user})
 
+    def _stage_changed(self, updates, original):
+        new_stage_id = updates.get('task', {}).get('stage', '')
+        old_stage_id = original.get('task', {}).get('stage', '')
+        return new_stage_id and new_stage_id != old_stage_id
```

Comment:
Sorry, already commented on the commit itself.
We need to cast to string so we know the comparison is reliable, usually the fetched item from Mongo will have the ID as ObjectId while the one from the request is string so will always return false.